### PR TITLE
Kill the process created by DataTarget.CreateSnapshotAndAttach

### LINKED ISSUE #925: CreateSnapshotAndAttach does not kill the created process
I would expect calling `Dispose` on the returned target to kill the suspended process created by `CreateSnapshotAndAttach`.

The problem can be seen with this simple sample:
```csharp
static void Main()
{
    for (int i = 0; i < 10; i++)
    {
        using var dt = DataTarget.CreateSnapshotAndAttach(Process.GetCurrentProcess().Id);
    }
}
 ```

 If you break on the last line you can see that the 10 processes are still there: 
 
image

While looking at the code, I noticed that adding in `WindowsProcessDataReader.Dispose`   

```csharp
Process.GetProcessById(ProcessId)?.Kill();
```
does fix the issue.

I'd be happy to propose a PR!

### PR DESCRIPTION
**Pull request by jairbubbles**
Closes #925

### CODE REVIEW

File: src/Microsoft.Diagnostics.Runtime/src/DataReaders/Windows/WindowsProcessDataReader.cs [hunk 1]
```diff
@@ -97,6 +97,18 @@ private void Dispose(bool _)
...
                     WindowsFunctions.NativeMethods.CloseHandle(_process);
 
+                if (_originalPid != 0)
+                {
+                    try
+                    {
+                        // Kill the cloned process as we don't need it anymore
+                        Process.GetProcessById(ProcessId).Kill();
+                    }
+                    catch (Win32Exception)
```
Comment (leculver) [inline src/Microsoft.Diagnostics.Runtime/src/DataReaders/Windows/WindowsProcessDataReader.cs:107]: This should be a blanket `catch {}` statement.

Usually I don't like adding those, but the original reason that block was removed is that exceptions were escaping this exact `catch (Win32Exception)` block and making it impossible to properly have a `using DataTarget dt = ...` statement without wrapping the whole thing in a try/catch.

Comment (leculver) [reply src/Microsoft.Diagnostics.Runtime/src/DataReaders/Windows/WindowsProcessDataReader.cs:107]: Or better yet, something like this (or similar):

``` C#
catch (Exception e)
{
    Trace.WriteLine($"Unable to kill the target process: {e.ToString()}");
}
```

**Review by leculver — CHANGES_REQUESTED** [review]
Interesting...I didn't realize the original change left hanging processes.

There seems to be some odd state where the process is gone by the time we get to `Kill` and sometimes not...  In either case, adding this back as it was originally causes issues.  We need to catch all possible exceptions coming out of GetProcessById/Kill with a blanket catch block to ensure `Dispose` doesn't throw.

File: src/Microsoft.Diagnostics.Runtime/src/DataReaders/Windows/WindowsProcessDataReader.cs (hunk 1 above)
Comment (jairbubbles) [reply src/Microsoft.Diagnostics.Runtime/src/DataReaders/Windows/WindowsProcessDataReader.cs]: Done!

**Review by leculver — APPROVED** [review]

**Comment by leculver** [comment]
LGTM!  I should have a new version released early next week with this fix.

**Comment by ltrzesniewski** [comment]
Shouldn't the `Process` object be disposed even though it's killed?

**Comment by jairbubbles** [comment]
That's a good remark @ltrzesniewski! I discovered recently that `Process`  is disposable but I've never seen patterns where we dispose it like that:
```
using (Process.Start())
{
}
```

I guess we rely on the finalizer.

File: src/Microsoft.Diagnostics.Runtime/src/DataReaders/Windows/WindowsProcessDataReader.cs [hunk 2]
```diff
@@ -97,6 +97,19 @@ private void Dispose(bool _)
                 if (_process != IntPtr.Zero)
                     WindowsFunctions.NativeMethods.CloseHandle(_process);
 
+                if (_originalPid != 0)
+                {
+                    try
+                    {
+                        // Kill the cloned process as we don't need it anymore
+                        Process.GetProcessById(ProcessId).Kill();
```
[the remark this answers is not in the document]
Comment (sungam3r) [reply src/Microsoft.Diagnostics.Runtime/src/DataReaders/Windows/WindowsProcessDataReader.cs:105]: ok

**Pull request merged**